test/svd.jl

```diff
@@ -11,17 +11,23 @@ using MatrixAlgebraKit: TruncationKeepAbove, diagview
     @testset "size ($m, $n)" for n in (37, m, 63)
         k = min(m, n)
         if LinearAlgebra.LAPACK.version() < v"3.12.0"
-            algs = (LAPACK_DivideAndConquer(), LAPACK_QRIteration(), LAPACK_Bisection())
+            algs = (LAPACK_DivideAndConquer(), LAPACK_QRIteration(), LAPACK_Bisection(),
+                    LAPACK_DivideAndConquer, :LAPACK_DivideAndConquer)
         else
             algs = (LAPACK_DivideAndConquer(), LAPACK_QRIteration(), LAPACK_Bisection(),
-                    LAPACK_Jacobi())
+                    LAPACK_Jacobi(), LAPACK_DivideAndConquer, :LAPACK_DivideAndConquer)
         end
         @testset "algorithm $alg" for alg in algs
             n > m && alg isa LAPACK_Jacobi && continue # not supported
             minmn = min(m, n)
             A = randn(rng, T, m, n)
 
-            U, S, Vᴴ = svd_compact(A; alg)
+            if VERSION < v"1.11"
+                # This is type unstable on older versions of Julia.
+                U, S, Vᴴ = svd_compact(A; alg)
+            else
+                U, S, Vᴴ = @constinferred svd_compact(A; alg=($alg))
+            end
```

Comment:
This is the case I mentioned that isn't type stable in older versions of Julia. I'd vote for letting this slide and telling users to use a newer versions of Julia if they care about type stability, let me know what you think.

Reply:
I'm assuming that this is also just some strange interplay with the non-heterogeneous tuple of algs, so this is more of a test artifact than a real problem. Definitely okay with letting this slide.